L1Trigger/TrackFindingTMTT/src/TP.cc

```diff
@@ -0,0 +1,192 @@
+
+  void TP::fillUse() {
+    constexpr bool useOnlyInTimeParticles = false;
+    constexpr bool useOnlyTPfromPhysicsCollisionFalse = false;
+    // Use looser cuts here those those used for tracking efficiency measurement.
+    // Keep only those TP that have a chance (allowing for finite track resolution) of being reconstructed as L1 tracks. L1 tracks not matching these TP will be defined as fake.
+
+    // Include all possible particle types here, as if some are left out, L1 tracks matching one of missing types will be declared fake.
+    static const vector<int> genPdgIdsAllUnsigned = {11, 13, 211, 321, 2212};
+    vector<int> genPdgIdsAll;
+    for (unsigned int i = 0; i < genPdgIdsAllUnsigned.size(); i++) {
+      genPdgIdsAll.push_back(genPdgIdsAllUnsigned[i]);
+      genPdgIdsAll.push_back(-genPdgIdsAllUnsigned[i]);
+    }
+
+    // Range big enough to include all TP needed to measure tracking efficiency
+    // and big enough to include any TP that might be reconstructed for fake rate measurement.
+    constexpr float ptMinScale = 0.7;
+    const float ptMin = min(settings_->genMinPt(), ptMinScale * settings_->houghMinPt());
+    constexpr double ptMax = 9.9e9;
+    const float etaExtra = 0.2;
+    const float etaMax = max(settings_->genMaxAbsEta(), etaExtra + std::abs(settings_->etaRegions()[0]));
+    constexpr double fixedVertRcut = 10.;
+    constexpr double fixedVertZcut = 35.;
+
+    static const TrackingParticleSelector trackingParticleSelector(ptMin,
+                                                                   ptMax,
+                                                                   -etaMax,
+                                                                   etaMax,
+                                                                   max(fixedVertRcut, settings_->genMaxVertR()),
+                                                                   max(fixedVertZcut, settings_->genMaxVertZ()),
+                                                                   0,
+                                                                   useOnlyTPfromPhysicsCollisionFalse,
+                                                                   useOnlyInTimeParticles,
+                                                                   true,
+                                                                   false,
+                                                                   genPdgIdsAll);
+
+    use_ = trackingParticleSelector(*trackingParticlePtr_);
+  }
+
+  //=== Check if this tracking particle can be used to measure the L1 tracking efficiency.
+
+  void TP::fillUseForEff() {
+    useForEff_ = false;
+    if (use_) {
+      constexpr bool useOnlyInTimeParticles = true;
+      constexpr bool useOnlyTPfromPhysicsCollision = true;
+      constexpr double ptMax = 9.9e9;
+      static const TrackingParticleSelector trackingParticleSelector(settings_->genMinPt(),
```

Comment:
remaining static variable

Reply:
Since "static const" data is thread safe, and following the discussion leading to https://github.com/cms-sw/cmssw/pull/29381#issuecomment-628889782 , we propose to leave this code unchanged.